integration-tests/modules/__tests__/product/admin/products.spec.ts

```diff
@@ -0,0 +1,200 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreateProductDTO,
+  IPricingModuleService,
+  IProductModuleService,
+  PriceListStatus,
+  PriceListType,
+  ProductDTO,
+  ProductVariantDTO,
+} from "@medusajs/types"
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import { createAdminUser } from "../../../../helpers/create-admin-user"
+import { createVariantPriceSet } from "../../../helpers/create-variant-price-set"
+
+jest.setTimeout(50000)
+
+const env = { MEDUSA_FF_MEDUSA_V2: true }
+const adminHeaders = { headers: { "x-medusa-access-token": "test_token" } }
+
+async function createProductsWithVariants(
+  productModule: IProductModuleService,
+  productsData: CreateProductDTO
+): Promise<[ProductDTO, ProductVariantDTO[]]> {
+  const { variants: variantsData, ...productData } = productsData
+
+  const [product] = await productModule.create([productData])
+
+  const variantsDataWithProductId = variantsData?.map((variantData) => {
+    return { ...variantData, product_id: product.id }
+  })
+
+  const variants = variantsDataWithProductId
+    ? await productModule.createVariants(variantsDataWithProductId)
+    : []
+
+  return [product, variants]
+}
+
+medusaIntegrationTestRunner({
+  env,
+  testSuite: ({ dbConnection, getContainer, api }) => {
+    describe("Admin: Products API", () => {
+      let appContainer
+      let product
+      let product2
+      let product3
+      let variant
+      let variant2
+      let variant3
+      let pricingModule: IPricingModuleService
+      let productModule: IProductModuleService
+
+      beforeAll(async () => {
+        appContainer = getContainer()
+        pricingModule = appContainer.resolve(ModuleRegistrationName.PRICING)
+        productModule = appContainer.resolve(ModuleRegistrationName.PRODUCT)
+      })
+
+      beforeEach(async () => {
+        await createAdminUser(dbConnection, adminHeaders, appContainer)
+      })
+
+      describe("GET /admin/products/:id", () => {
```

Comment:
I think having these tests in the module is fine, but I'll probably move them (or remove them if they are duplicate) once I finish with enabling the `api` tests, just FYI

Reply:
Are both v1 and v2 test suites ready to be run within api? I couldn't see V2 run in the pipeline, so i opted to still keep it here. But all good to move when its ready 👍🏻

Reply:
@riqwan they are not ran in the pipeline yet, but they can be run manually while you are doing development. You can see my message in Slack from few days ago on how to do it 😄

Reply:
Yes, got that. I def want this running on the pipeline, thats why I kept it in modules.